mmv1/third_party/terraform/tpgresource/utils.go

```diff
@@ -237,6 +238,87 @@ func ExpandStringMap(d TerraformResourceData, key string) map[string]string {
 	return ConvertStringMap(v.(map[string]interface{}))
 }
 
+// SortStringsByConfigOrder takes a slice of map[string]interface{} from a TF config
+// and API data, and returns a new slice containing the API data, reorderd to match
+// the TF config as closely as possible (with new items at the end of the list.)
+func SortStringsByConfigOrder(configData, apiData []string) ([]string, error) {
+	configOrder := map[string]int{}
+	for index, item := range configData {
+		_, ok := configOrder[item]
+		if ok {
+			return nil, fmt.Errorf("configData element at %d has duplicate value `%s`", index, item)
+		}
+		configOrder[item] = index
+	}
+
+	apiSeen := map[string]struct{}{}
+	byConfigIndex := map[int]string{}
+	newElements := []string{}
+	for index, item := range apiData {
+		_, ok := apiSeen[item]
```

Comment:
I can't see anywhere where `apiSeen` has key:value pairs added to it - should this be happening at the end of this for loop block?

Reply:
Fixed in 0e3ccd403